### PR TITLE
Test controllers book

### PR DESCRIPTION
**Pull request by gussiciliano**
### Sumary
- add Tests for books_controller.rb

### Trello Card
https://trello.com/c/eNMn61Bo/25-crear-tests-para-rutas-index-y-show-de-book

### CODE REVIEW

File: app/serializers/book_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class BookSerializer < ActiveModel::Serializer
+  attributes :genre, :author, :image, :title, :publisher, :year
+end
```
Comment (Dinuuu) [inline app/serializers/book_serializer.rb]: Its convention to enter a new line at the end of file

File: spec/factories/books_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FactoryGirl.define do
+  factory :book do
+    genre     Faker::Book.genre
+    author    Faker::Book.author
+    image     "an image"
+    title     Faker::Book.title
+    publisher Faker::Book.publisher
+    year      "2016"
+  end
+end
```
Comment (Dinuuu) [inline spec/factories/books_spec.rb]: idem

File: spec/factories/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+  factory :user do
+    first_name Faker::Name.first_name
+    last_name  Faker::Name.last_name 
+    email      Faker::Internet.email
+    password   Faker::Internet.password
+  end
+end
```
Comment (Dinuuu) [inline spec/factories/users_spec.rb]: idem

File: spec/shared_context.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+RSpec.shared_context 'Authenticated User' do
+  let(:user) { create(:user) }
+  let!(:access_data) { AuthenticableEntity.generate_access_token(user) }
+  let!(:access_token) { access_data[:token] }
+  let!(:renew_id) { access_data[:renew_id] }
+
+  before(:each) do
+    request.headers['Authorization'] = access_token
+  end
+end
```
Comment (Dinuuu) [inline spec/shared_context.rb]: idem

File: app/serializers/book_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class BookSerializer < ActiveModel::Serializer
+  attributes :genre, :author, :image, :title, :publisher, :year
```
Comment (Dinuuu) [inline app/serializers/book_serializer.rb]: Check that the image is not rendering well

File: app/serializers/book_serializer.rb (hunk 1 above)
Comment (gussiciliano) [inline app/serializers/book_serializer.rb]: Oks

File: spec/factories/books_spec.rb (hunk 2 above)
Comment (gussiciliano) [inline spec/factories/books_spec.rb]: Oks

File: spec/factories/users_spec.rb (hunk 3 above)
Comment (gussiciliano) [inline spec/factories/users_spec.rb]: Oks

File: spec/shared_context.rb (hunk 4 above)
Comment (gussiciliano) [inline spec/shared_context.rb]: Oks

File: app/serializers/book_serializer.rb (hunk 5 above)
Comment (gussiciliano) [inline app/serializers/book_serializer.rb]: Oks

File: app/serializers/book_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class BookSerializer < ActiveModel::Serializer
+  attributes :id, :author, :title, :image, :publisher, :year, :genre, :created_at, :updated_at
+
+  def image
+    "url:{" + object.image + "}"
```
Comment (Dinuuu) [inline app/serializers/book_serializer.rb]: The same I point out in the books controller PR I don't think this is working

File: app/serializers/book_serializer.rb [hunk 7]
```diff
@@ -1,7 +1,9 @@
 class BookSerializer < ActiveModel::Serializer
-  attributes :id, :author, :title, :image, :publisher, :year, :genre, :created_at, :updated_at
+
```
Comment (Dinuuu) [inline app/serializers/book_serializer.rb]: Delete this empty lines

**Pull request merged**